converter/codec_test.go

```diff
@@ -138,47 +138,20 @@ func assertEncodingDataConverter(t *testing.T, data interface{}) {
 	}
 
 	// Check that it's ignored if too small (which all params given are)
-	zlibIgnoreMinConv := converter.NewEncodingDataConverter(
+	zlibIgnoreMinConv := converter.NewCodecDataConverter(
 		defaultConv,
-		converter.NewZlibEncoder(converter.ZlibEncoderOptions{}),
+		converter.NewZlibCodec(converter.ZlibCodecOptions{}),
 	)
 	require.NoError(t, err)
 	compUnderMinPayload, err := zlibIgnoreMinConv.ToPayload(data)
 	require.NoError(t, err)
 	require.True(t, proto.Equal(uncompPayload, compUnderMinPayload))
 }
 
-func TestEncodingDataConverterClone(t *testing.T) {
```

Comment:
I have removed the cloning from `ToPayload(s)` because I believe the clone is only required for the `FromPayload(s)` cases where we must not mutate the `Payload` passed in. I believe it is safe to mutate `Payload` we receive from our parent `DataConverter` as those would not yet have been seen by any other code. Let me know if this is not correct.